hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java

```diff
@@ -47,31 +51,39 @@ public record TransactionMetadata(
         @Nullable HederaKey payerKey,
         @NonNull List<HederaKey> requiredNonPayerKeys,
         @Nullable Object handlerMetadata,
+        @Nullable TransactionSignature payerSignature,
+        @NonNull List<TransactionSignature> otherSignatures,
         @NonNull List<ReadKeys> readKeys) {
 
     public TransactionMetadata {
         requireNonNull(status);
         requireNonNull(requiredNonPayerKeys);
+        requireNonNull(otherSignatures);
         requireNonNull(readKeys);
     }
 
     public TransactionMetadata(
-            @NonNull final PreHandleContext context, @NonNull final List<ReadKeys> readKeys) {
+            @NonNull final PreHandleContext context,
+            @Nullable final TransactionSignature payerSignature,
```

Comment:
Naïve question: Is payer always singular?  Do we not support multiple payment accounts on a transaction?

Reply:
I believe we do support multiple payment accounts, but there is always a single payer, that is stored in a special place and handled differently. I think for example, the payer has to pay, if the transaction fails because of a user error.